MSProject/ViewControllers/DetailBook/WBDetailBookViewController.swift

```diff
@@ -88,7 +77,20 @@ class WBDetailBookViewController: UIViewController {
         
         _detailHeaderView.wishlistButton.reactive.controlEvents(.touchUpInside)
             .observeValues { _ in
-            self.addToWishlist()
+            
+                MBProgressHUD.showAdded(to: self._view, animated: true)
+
+                self.viewModel.wishBook(book: self.bookViewModel.book).startWithResult { [unowned self] result in
+                    switch result {
+                    case .success:
+                        self.bookViewModel.wished = true
+                        self._detailHeaderView.setup(with: self.bookViewModel)
+                    case .failure(let error):
+                        self.showAlert(message: error.localizedDescription)
+                    }
+                    MBProgressHUD.hide(for: self._view, animated: true)
```

Comment:
If i'm not mistaken, if you try to add a book to your wishlist and while it's loading you go back to the home view, when the request arrives the app will crash, since self doesn't exist anymore. To avoid that you have 2 options: you can use weak instead of unowned and then check if self exists. Or my favorite option, you can use the .take(during:) operator on the signal, this operator binds the signal life to an objects life, in this case you want the signal to get disposed when your controller gets deinited, so you can do:
`self.viewModel.wishBook(book: self.bookViewModel.book).take(during: self.reactive.lifetime).startWithResult`

Reply:
Shall I do that implementation on every startWithResult I guess? bad MBProgressHUD that dont disable the navigation =P

Reply:
yeah, it's usually a good idea to add it just in case, you can make a SignalProducer extension and create your own startWithResult function that has an extra parameter of type Lifetime if you don't want to add it to every request (wouldn't be a bad idea for wolmo reactive).
It would be easier if progresshud disabled the navigation, but unfortunately that usually not the desired behavior, as a user you probably want to be able to go back even if the view didn't finish loading (maybe you entered the view by accident or something).